### ISSUE TITLE
Moving back from post title activates select mode

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#49142]
While working a pr that [Updates the behavior of the top toolbar fixed setting](https://github.com/WordPress/gutenberg/pull/47277) I found a bug with keyboard navigation which inadvertently activates the select mode inside the post editor.

To reproduce:

1. In a post editor
2. Notice in the top toolbar the icon next to the inserter is the pencil icon (edit)
3. Focus (click in) the post title
4. Then press Shift + Tab on the keyboard twice
5. Notice in the top toolbar the icon next to the inserter is the hollow arrow icon (select)
6. Notice nothing is actually focused (focus loss)
7. Then press Shift + Tab on the keyboard once
8. Notice the focus is now on the top right options menu

This is wrong on several points I think:

- there should be no focus loss, and there is focus loss for 2 of the three steps
- the select mode has nothing to do with attempting to navigate back into the editor header with the keyboard

What happens focus wise:

- focus h1 of post title
- shift tab focus the body of the iframe - focus loss
- shift tab focus the html element of the iframe (!) (turns on navigation mode)  - focus loss
- shift tab focus the body of the iframe, but also moves focus in the parent of the iframe to the options menu
- also for some reason there is a `SET_EDITOR_MODE` action dispatched!

### Walkthrough

https://user-images.githubusercontent.com/107534/225679835-2330562c-2c2e-4f01-ae29-7b3a68548daa.mp4

**Comment by talldan** [comment]
Strange one. It might be worth testing if it happens with the iframe disabled. It seems to happen exactly as focus leaves the editor canvas iframe.

**Comment by talldan** [comment]
Closing this as Select mode has since been removed
